### PR TITLE
[BugFix] forbid MV rewrite when query has tablet or partition hints

### PR DESCRIPTION
**Pull request by satanson**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [x] 2.4
  - [x] 2.3

### CODE REVIEW

**Comment by Seaven** [comment]
add some ut?

**Review by Seaven — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/logical/LogicalOlapScanOperator.java [hunk 1]
```diff
@@ -38,6 +38,7 @@ public final class LogicalOlapScanOperator extends LogicalScanOperator {
     private final long selectedIndexId;
     private final List<Long> selectedPartitionId;
     private final PartitionNames partitionNames;
+    private final boolean hasHintsPartitionNames;
```
[the remark this answers is not in the document]
Comment (satanson) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/logical/LogicalOlapScanOperator.java:41]: The original LogicalScanOperators's partitionNames is NULL, but later, MV rule will fill this field, so it is not used to dictate whether there is partition hint or not.

**Review by ABingHuang — DISMISSED** [review]
LGTM

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by satanson** [comment]
run starrocks_fe_unittest

**Comment by satanson** [comment]
run starrocks_fe_unittest

**Review by kangkaisen — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 17 / 18 (94.44%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/mv/MaterializedViewRule.java|1|2|50.00%|[122]|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/RelationTransformer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/TableRelation.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalOlapScanOperator.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/rule/BaseMaterializedViewRewriteRule.java|8|8|100.00%|[]|

**Comment by satanson** [comment]
run starrocks_admit_test

**Comment by satanson** [comment]
@mergifyio backport branch-2.3

**Comment by satanson** [comment]
@mergifyio backport branch-2.4

**Comment by satanson** [comment]
@mergifyio backport branch-2.5

**Comment by satanson** [comment]
@mergifyio backport branch-3.0

**Pull request merged**